src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,981 @@
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestConstrainedTypeParameter()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    void M<T>(T s) where T : class
+    {
+        [|if ((object)s == null)
+            throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    void M<T>(T s!!) where T : class
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestUnconstrainedTypeParameter()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode =
+@"using System;
+
+class C
+{
+    void M<T>(T s)
+    {
+        [|if ((object)s == null)
+            throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode =
+@"using System;
+
+class C
+{
+    void M<T>(T s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestComplexExpr()
+        {
+            var testCode = @"using System;
+
+class C
+{
+    void M(string[] s)
+    {
+        if ((object)s[0] == null)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestComplexExpr2()
+        {
+            var testCode = @"using System;
+
+class C
+{
+    void M(string s1, string s2)
+    {
+        if (s1 == null || s2 == null)
+            throw new ArgumentNullException(nameof(s1));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotOnDefault()
+        {
+            var testCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ((object)s == default)
+            throw new ArgumentNullException(nameof(s));
```

Comment:
interesting.  i think this is ok for now.

Reply:
This was brought in inadvertently because I based my work on the "use 'is null' instead of '== null'" feature. I would be open to making any null constant do the right thing here in a follow up.